MapboxNavigation/SpeedLimit.swift

```diff
@@ -0,0 +1,159 @@
+import Foundation
+import MapboxDirections
+
+/// :nodoc:
+@objc(MBSpeedLimitSign)
+open class SpeedLimitSign: UIStackView {
+    var labels: [SpeedLimitLabel] = []
+    
+    let defaultText = NSLocalizedString("SPEED_LIMIT", bundle: .mapboxNavigation, value: "Speed Limit", comment: "Speed limit sign main text.")
+    var defaultTextSplit: [String] {
+        return defaultText.components(separatedBy: " ")
+    }
+    
+    let paragraphStyle = NSMutableParagraphStyle()
+    let fontSize: CGFloat = 14.0
+    let strings = NSMutableAttributedString()
+    let numberFormater = NumberFormatter()
+    let padding: CGFloat = 16
+    
+    var region: SpeedLimitSignRegionType!
+    
+    var speedLimit: SpeedLimit? {
+        didSet {
+            guard let speedLimit = speedLimit else { return }
+            guard let speedString = numberFormater.string(from: speedLimit.value as NSNumber) else { return }
+            labels.removeAll()
+            
+            // Only include the string `SPEED LIMIT` for .unitedStates.
+            if region == .unitedStates {
+                let defaultLabels = defaultTextSplit.map { (string: String) -> SpeedLimitLabel in
+                    let label: SpeedLimitLabel = .forAutoLayout()
+                    label.attributedText = NSMutableAttributedString(string: string.uppercased(), attributes: [
+                        .font: UIFont.systemFont(ofSize: fontSize * 0.9),
+                        .paragraphStyle: paragraphStyle
+                        ])
+                    return label
+                }
+                labels.append(contentsOf: defaultLabels)
+            }
+            
+            let speedLabel: SpeedLimitLabel = .forAutoLayout()
+            speedLabel.attributedText = NSMutableAttributedString(string: speedString, attributes: [
+                .font: UIFont.boldSystemFont(ofSize: fontSize * 2)
+                ])
+            if region == .world {
+                speedLabel.adjustsFontSizeToFitWidth = true
+                speedLabel.numberOfLines = 1
+                speedLabel.minimumScaleFactor = 0.5
+                speedLabel.allowsDefaultTighteningForTruncation = true
+            }
+            labels.append(speedLabel)
+            
+            let unitLabel: SpeedLimitLabel = .forAutoLayout()
+            unitLabel.attributedText = NSMutableAttributedString(string: speedLimit.unit.localizedSpeedUnit, attributes: [
+                .font: UIFont.boldSystemFont(ofSize: fontSize * 0.7)
+                ])
+            labels.append(unitLabel)
+            
+            updateStackView(with: labels)
+        }
+    }
+    
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        commonInit()
+    }
+    
+    required public init(coder: NSCoder) {
+        super.init(coder: coder)
+        commonInit()
+    }
+    
+    func commonInit() {
+        axis = .vertical
+        alignment = .center
+        spacing = 0.5
+        if region == .world {
```

Comment:
I'm not excited how this class is turning out. I think it might make more sense to extract these if statements out further. If @mapbox/navigation-ios has any suggestions, im all ears.